app/controllers/users_controller.rb

```diff
@@ -54,11 +60,32 @@ def set_user
 
     # Only allow a list of trusted parameters through.
     def user_params
-      params.require(:user).permit([:display_name, :full_name, :family_name, :orcid, :email_messages_enabled])
+      @user_params ||= params.require(:user).permit([:display_name, :full_name, :family_name, :orcid, :email_messages_enabled, { collections_with_messaging: {} }])
     end
 
     def can_edit?
       return true if current_user.id == @user.id
       current_user.super_admin?
     end
+
+    def parameter_enables_messaging?(form_value)
+      form_value.to_s == COLLECTION_MESSAGING_ENABLED
+    end
+
+    def update_collections_with_messaging
+      if user_params.key?(:collections_with_messaging)
+        extracted = user_params.extract!(:collections_with_messaging)
+        collections_with_messaging = extracted[:collections_with_messaging]
+
+        collections_with_messaging.each_pair do |collection_id, param|
+          selected_collection = Collection.find_by(id: collection_id)
```

Comment:
Is each `find_by` a separate DB hit? If we know is a small number, then probably fine, but if it could be large, consider finding them all with one query. It looks like there used to be a [build-in](https://stackoverflow.com/questions/9650205/rails-find-all-with-conditions) syntax for this, but we'd now use `where()`.

Reply:
No, I apologize, `collections_with_messaging = extracted[:collections_with_messaging]` yields a `Hash`, which itself is parsed from the HTTP request `params`